frontend/public/components/build-pipeline.scss

```diff
@@ -0,0 +1,334 @@
+  font-size: ($pipeline-font-base - 1);
+}
+.build-pipeline__animation-line {
+  background: $pipeline-pending-color;
+  height: $pipeline-line-height;
+  position: relative;
+  width: $pipeline-progress-line;
+  &:before {
+    animation: build-progress-line $pipeline-line-animation-time ease-in forwards;
+    content: '';
+    height: 100%;
+    position: absolute;
+    width: 0;
+  }
+}
+.build-pipeline__animation-circle {
+  background: $pipeline-pending-color;
+  border-radius: $pipeline-circle-radius;
+  height: $pipeline-circle-diameter;
+  margin-top: (-($pipeline-circle-diameter / 2) - ($pipeline-line-height / 2));
+  position: relative;
+  transform: rotate(-90deg);
+  width: $pipeline-circle-diameter;
+  &:after {
+    animation: pipeline-stage-fadeIcon $pipeline-icon-animation-time ($pipeline-line-grow-animation-time + $pipeline-semi-circle-animation-time + ($pipeline-inner-circle-animation-time * 2)) linear forwards;
+    color: $pipeline-inner-circle-color;
+    font-family: 'FontAwesome';
+    font-size: $pipeline-font-base;
+    left: 50%;
+    opacity: 0;
+    position: absolute;
+    top: 50%;
+    transform: translate(-50%, -50%) rotate(90deg);
+  }
+}
+
+%circle-clip {
+  position: absolute;
+  z-index: -9;
+}
+
+%circle-clip-before {
+  border-radius: $pipeline-circle-radius;
+  content: '';
+  height: $pipeline-circle-diameter;
+  position: absolute;
+  transform: rotate(0);
+  width: $pipeline-circle-diameter;
+}
+
+.build-pipeline__circle-clip1 {
+  @extend %circle-clip;
+  clip: rect(0, $pipeline-circle-diameter, $pipeline-circle-diameter, $pipeline-circle-radius);
+  &:before {
+    @extend %circle-clip-before;
+    animation: build-progress $pipeline-semi-circle-animation-time $pipeline-line-animation-time linear forwards;
+    clip: rect(0, $pipeline-circle-radius, $pipeline-circle-diameter, 0);
+  }
+}
+.build-pipeline__circle-clip2 {
+  @extend %circle-clip;
+  clip: rect(0, $pipeline-circle-radius, $pipeline-circle-diameter, 0);
+  &:before {
+    @extend %circle-clip-before;
+    animation: build-progress $pipeline-semi-circle-animation-time ($pipeline-semi-circle-animation-time + $pipeline-line-animation-time) linear forwards;
+    clip: rect(0, $pipeline-circle-diameter, $pipeline-circle-diameter, $pipeline-circle-radius);
+  }
+}
+.build-pipeline__circle-inner {
+  animation: pipeline-stage-fadeOut $pipeline-inner-circle-animation-time ($pipeline-line-animation-time + $pipeline-circle-animation-time) linear forwards;
+  background-color: $pipeline-inner-circle-color;
+  border-radius: $pipeline-circle-radius;
+  height: ($pipeline-circle-diameter - $pipeline-circle-border-width * 2);
+  left: $pipeline-circle-border-width;
+  position: absolute;
+  top: $pipeline-circle-border-width;
+  width: ($pipeline-circle-diameter - $pipeline-circle-border-width * 2);
+}
+.build-pipeline__circle-inner-fill {
+  border-radius: 50%;
+  box-sizing: border-box;
+  height: 100%;
+  opacity: 0;
+  width: 100%;
+}
```

Comment:
Just an observation, since we've been kicking around the topic.  Most of the other existing SCSS files seem to hover around 50 lines or less.  There are a couple in the range of 100.  

350 lines is pretty big.

Reply:
Per the file size, I'm wondering if it would be good to break the elements down.  I'm partial to the pattern I see in a lot of React apps where you have something like:

```bash 
/public
  /components
    /build-pipeline
      |_ build-pipeline.js 
      |_ build-stage.js 
      |_ etc.
```
And do an 

```JavaScript
// build-pipeline.js
import './build-pipeline.scss';

// build-stage.js
import './build-stage.scss';

// etc.
```
But the current pattern is more or less one big file list in `public/components` so I don't necessarily want to go out of the norm.

Any thoughts? @rhamilto @sg00dwin 
As well as @alecmerdler @spadgett since its a structure question.

Otherwise, I'm going to remove the `[WIP]` because we might be close to settled.

Reply:
@benjaminapetersen Well, we can split it up to make it smaller, though a cursory look at the code, if you pull the rest out into a `build-pipeline.scss` partial,  the majority of code would still be in `build-stage.scss`. Probably would still be 200 lines.

Reply:
Yeah I think our verbals on this indicated we don't want to split it up at this point.